### ISSUE TITLE
installcheck failed tests

### ISSUE THREAD

**Issue by antolikjan** [nest/nest-simulator#237]
Nest 2.10.0, intel compiler suite, RedHat version 6.7

Configure command with which compilation was done:

CXX=mpiicpc ./configure --with-mpi --disable-openmp --without-openmp -- --with-debug="-DMPICH_IGNORE_CXX_SEEK"

[reports.zip](https://github.com/nest/nest-simulator/files/138234/reports.zip)

**Comment by antolikjan** [comment]
We have progressed with the installation of the nest on the system. After using following configuration options we now get only 3 faild tests once running installcheck. Could you please let us know if any of them is critical?

configure CC=mpiicc CFLAGS="-openmp -mt_mpi -O3" CXX=mpiicpc CXXFLAGS="-openmp -mt_mpi -O3" LDFLAGS="-L/.../mkl/lib/intel64 -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -L/.../compiler/lib/intel64 -liomp5"

[config.log.tar.gz](https://github.com/nest/nest-simulator/files/148362/config.log.tar.gz)
[reports.zip](https://github.com/nest/nest-simulator/files/148361/reports.zip)

**Comment by heplesser** [comment]
@antolikjan From scanning the logs, it seems the following tests failed:

- unittests/test_dcgen_versus_I_e.sli
- test_ThreadsFindConnections
- test_ThreadsGetEvents

They should all pass. I have access to a system with the intel compiler suite and can try there within a few days. It is a bit tricky to make modify the tests so that they tell you what is going wrong.

Btw, why do you need the `-DMPICH_IGNORE_CXX_SEEK` switch?

**Comment by antolikjan** [comment]
@heplesser Thank you, that would be extremely helpful. Regarding the DMPICH_IGNORE_CXX_SEEK flag: that was a suggestion from nest FAQ to avoid errors we were getting, but note that the configure command in my second post doesn't use it anymore.

**Comment by heplesser** [comment]
@antolikjan Could you compile NEST using the gcc compiler suite instead of Intel on the cluster you are using, so that you would (hopefully) have a working setup until we have solved the problems with the Intel compiler suite?

**Comment by heplesser** [comment]
@antolikjan The configure line you reported four days ago does not contain `--with-mpi`. Was that intentional or an oversight? It would explain several observations that you reported in #238, because without `--with-mpi`, NEST will be compiled without MPI support, even though you are using the MPI compiler wrappers. I am currently testing with Intel compilers and will get back to you soon.

**Comment by heplesser** [comment]
@antolikjan I have now compiled NEST without MPI and without MKL as follows

```
 ../src/configure CC=icc CFLAGS="-openmp -O3" CXX=icpc CXXFLAGS="-openmp -O3" --prefix=`pwd`/install
```

All tests pass (except MPI tests, which are skipped). The three tests that failed when you tested also passed. Since those tests are all independent of MPI, I assume something else went wrong. As a next step, I will try building serially but with MKL.

**Comment by heplesser** [comment]
@antolikjan I have now compiled a serial (i.e. thread-parallel, no-mpi) version with MKL using

```
../src/configure CC=icc CFLAGS="-openmp -O3" CXX=icpc CXXFLAGS="-openmp -O3" LDFLAGS="-L/.../mkl/lib/intel64 -lmkl_intel_lp64 -lmkl_core -lmkl_intel_thread -lpthread -lm -L/.../compiler/lib/intel64 -liomp5" --prefix=`pwd`/install
```

All tests pass.

Btw, I am using the Intel compilers version 14.0.2 and MKL 11.1 on a cluster running Rocks Linux.

**Comment by heplesser** [comment]
@antolikjan Here now the results from my experiments with Intel compilers, MKL and MPI. I compiled with (broken across a few lines here for better readability)
```
../src/configure 
  CC=mpiicc CFLAGS="-openmp -mt_mpi -O3" 
  CXX=mpiicpc CXXFLAGS="-openmp -mt_mpi -O3 -DMPICH_IGNORE_CXX_SEEK" 
  LDFLAGS="-L/.../mkl/lib/intel64 -lmkl_intel_lp64 -lmkl_core -lmkl_intel_thread 
             -lpthread -lm -L/.../compiler/lib/intel64 -liomp5" 
   --prefix=/home/plesser/NEST_Intel/nest-2.10.0/bld_mpi_mkl_nopy/install 
   --with-mpi --without-python
```
Note that I had to deselect Python, because I otherwise got a conflict between MPI libs with and without OpenMP support when linking `pynestkernel.so`.

The resulting executable passes all unit and regression tests. The MPI tests fail randomly though. I have not been able to confirm this entirely, but I suspect that this is a problem with the NEST MPI testing mechanism, which is long due for an overhaul.

To test that NEST is working correctly, I ran slightly modified versions of `brunel-2000_newconnect.sli` and `hpc_benchmark.sli` with 16 virtual processes distributed across different numbers of MPI processes and threads. They yielded identical results, as expected (I will add the scripts I used in the next post).

As a comparison, I also built NEST 2.10.0 with the default gcc on the system (4.7.2) using OpenMPI 1.6.2 and the configure line
```
 ../src/configure --with-mpi --prefix=`pwd`/install
```
NEST builds with Python in this case and passes all tests. Running the hpc_benchmark with the resulting executable yielded identical results as with the Intel-compiled executable, but took about 50% longer.

I hope this helps so far. I will create a new issue about making NEST compile with Intel compilers and MPI even when using Python, and to put instructions on the website. The 50% performance gain would be worth some effort.
